tests/test_consdbClient.py

```diff
@@ -161,6 +164,38 @@ def test_schema(client):
     assert client.schema(instrument, table) == description
 
 
+@responses.activate
+def test_clean_token_url_response():
+    """Test tokens URL is cleaned when an error is thrown from requests
+    Use with pytest raises assert an error,
+    assert that url does not contain tokens
+    """
+    complex_client = ConsDbClient("https://usdf:v987wefVMPz@usdf-fake.slackers.stanford.edu/consdb")
+
+    obs_type = "exposure"
+    responses.post(
+        "https://usdf:v987wefVMPz@usdf-fake.slackers.stanford.edu/consdb/flex/bad_instrument/exposure/addkey",
+        status=404,
+    )
+    with pytest.raises(HTTPError, match="404") as error:
+        complex_client.add_flexible_metadata_key(
+            "bad_instrument", obs_type, "error", "int", "instrument error"
+        )
+
+    url = error.value.args[0].split()[-1]
+    sanitized = (
+        "https://us***:v9***@usdf-fake.slackers.stanford.edu/consdb/flex/bad_instrument/exposure/addkey"
+    )
+
+    assert "v987wefVMPz" not in url
+    assert url == sanitized
+
```

Comment:
If you really want coverage, you could also test the username/no password case, a password/no username case (although I don't know if that is actually legal), and a single-character username case (if someone uses 1 or 2 character passwords, that's their own problem...).